sdk/core/Azure.Core/src/ResponseError.cs

```diff
@@ -189,5 +197,48 @@ internal void Append(StringBuilder builder, bool includeRaw)
                 builder.Append(_element.GetRawText());
             }
         }
+
+        /// <inheritdoc />
+        public void Write(Utf8JsonWriter writer, ModelReaderWriterOptions options)
+        {
+            throw new NotSupportedException(NotSupportedExceptionMessage);
```

Comment:
I think for now we should leave the behavior as-is. I haven't been able to understand all the implications of changing this type to allow serialization, and given the accelerated timeline of this feature, it's not possible for me to do so. I don't feel comfortable having my name on something that is potentially unsafe and that I don't understand.

Reply:
This completely preserves the current behavior of this type, which was established 4 years ago.